### PR TITLE
TS: handle ochGetCommand with no arguments (read all)

### PR DESCRIPTION
**Pull request by dron0gus**
Get all livedata with one command with no arguments. Preparation for blockingFactor = infinity.

### CODE REVIEW

File: firmware/console/binary/tunerstudio.cpp [hunk 1]
```diff
@@ -686,12 +686,23 @@ int TunerStudio::handleCrcCommand(TsChannelBase* tsChannel, char *data, int inco
 
 	const uint16_t* data16 = reinterpret_cast<uint16_t*>(data);
 
-	uint16_t offset = data16[0];
-	uint16_t count = data16[1];
+	uint16_t offset = 0;
+	uint16_t count = 0;
+
+	if (incomingPacketSize >= 3) {
```
Comment (mck1117) [inline firmware/console/binary/tunerstudio.cpp:692]: You don't need this check, data16[0]/[1] are always guaranteed to be inside the write buffer even if they didn't get written this time.

Comment (mck1117) [reply firmware/console/binary/tunerstudio.cpp:692]: (doing it this way adds extra branches whose result is always ignored)

File: firmware/console/binary/tunerstudio.cpp [hunk 2]
```diff
@@ -686,12 +686,23 @@ int TunerStudio::handleCrcCommand(TsChannelBase* tsChannel, char *data, int inco
...
+	}
+	if (incomingPacketSize >= 5) {
+		count = data16[1];
+	}
 
 	switch(command)
 	{
 	case TS_OUTPUT_COMMAND:
+		if (incomingPacketSize == 1) {
+			/* Read command with no offset and size - read whole livedata */
```
Comment (mck1117) [inline firmware/console/binary/tunerstudio.cpp:703]: style here is `//` comments

File: firmware/console/binary/tunerstudio.cpp (hunk 1 above)
Comment (dron0gus) [reply firmware/console/binary/tunerstudio.cpp:692]: Yes. It is guaranteed by current design. Not by function definition.
It is safe to access this memory. But it is not safe to use values that can be random.
Good compiler will not insert branching here, just a few instructions with conditional execution.

**Pull request merged**